src/simulation/ApplyLoad.cpp

```diff
@@ -0,0 +1,266 @@
+        lm.getLastClosedLedgerNum() + 1, 0, wasmBytes, contractCodeLedgerKey,
+        std::nullopt);
+
+    closeLedger({uploadTx.second});
+
+    for (auto kvp : mTxGenerator.getAccounts())
+    {
+        auto salt = sha256("Load contract " + std::to_string(kvp.first));
+
+        auto createTx = mTxGenerator.createContractTransaction(
+            lm.getLastClosedLedgerNum() + 1, 0, contractCodeLedgerKey,
+            wasmBytes.size() + 160, salt, std::nullopt);
+        closeLedger({createTx.second});
+
+        auto instanceKey =
+            createTx.second->sorobanResources().footprint.readWrite.back();
+
+        TxGenerator::ContractInstance instance;
+        instance.readOnlyKeys.emplace_back(mLoadCodeKey);
+        instance.readOnlyKeys.emplace_back(instanceKey);
+        instance.contractID = instanceKey.contractData().contract;
+        mLoadInstances.emplace(kvp.first, instance);
+    }
+}
+
+void
+ApplyLoad::benchmark()
+{
+    auto& lm = mApp.getLedgerManager();
+    std::vector<TransactionFrameBasePtr> txs;
+
+    auto resources = multiplyByDouble(lm.maxLedgerResources(true),
+                                      TRANSACTION_QUEUE_SIZE_MULTIPLIER);
+
+    auto const& accounts = mTxGenerator.getAccounts();
+    std::vector<uint64_t> shuffledAccounts(accounts.size());
+    std::iota(shuffledAccounts.begin(), shuffledAccounts.end(), 0);
+    stellar::shuffle(std::begin(shuffledAccounts), std::end(shuffledAccounts),
+                     gRandomEngine);
+
+    for (auto accountIndex : shuffledAccounts)
+    {
+        auto it = accounts.find(accountIndex);
+        releaseAssert(it != accounts.end());
+
+        auto instanceIter = mLoadInstances.find(it->first);
+        releaseAssert(instanceIter != mLoadInstances.end());
+        auto const& instance = instanceIter->second;
+        auto tx = mTxGenerator.invokeSorobanLoadTransaction(
+            lm.getLastClosedLedgerNum() + 1, it->first, instance,
+            rust_bridge::get_write_bytes().data.size() + 160, std::nullopt);
+
+        if (!anyGreater(tx.second->getResources(false), resources))
+        {
+            resources -= tx.second->getResources(false);
+        }
+        else
+        {
+            for (size_t i = static_cast<size_t>(Resource::Type::OPERATIONS);
+                 i <= static_cast<size_t>(Resource::Type::WRITE_LEDGER_ENTRIES);
+                 ++i)
+            {
+                auto type = static_cast<Resource::Type>(i);
+                if (tx.second->getResources(false).getVal(type) >
+                    resources.getVal(type))
+                {
+                    CLOG_INFO(Perf, "Ledger {} limit hit during tx generation",
+                              Resource::getStringFromType(type));
+                }
+            }
+            break;
+        }
+
+        txs.emplace_back(tx.second);
+    }
+
+    closeLedger(txs);
+}
+
+double
+ApplyLoad::successRate()
+{
+    return mTxGenerator.GetApplySorobanSuccess().count() * 1.0 /
```

Comment:
The denominator should be the number of transactions we've tried to apply, right? This will catch a scenario where we're generating invalid transactions for whatever reason (currently invalid transactions will just be not accounted for at all).

Reply:
The generated transactions shouldn't be invalid, so I added a checkValid check when we generate them.